src/robot_dart/robot.cpp

```diff
@@ -1618,6 +1622,11 @@ namespace robot_dart {
         return detail::dof_data<15>(_skeleton, dof_names, _dof_map);
     }
 
+    Eigen::VectorXd Robot::constraint_forces(const std::vector<std::string>& dof_names) const
```

Comment:
What does the `constraint_forces` has to do with the detaching of sensor (that this PR handles)? If this is needed for something else, can you make a different PR? Thanks!

Reply:
We need it to compute the SPD+ (for position control) with a broken robot. I agree that this is not about "broken sensor", but we need broken sensors to handle damaged robots. I don't think few lines are worth a PR. We can change the commit message if you care.

Reply:
OK then. Then change the name of the PR (e.g. `Allow to "break" sensors + constraint forces`) or put in bold in the description something about this..